conformance/lang/expressions/additive-expr/decimal_addition.balt

```diff
@@ -0,0 +1,985 @@
+    io:println(a); // @output 2.500000000000000000000000000000000E+1743
+
+    a = a1 + a6;
+    io:println(a); // @output -2.500000000000000000000000000000000E+1743
+
+    a = a1 + 10;
+    io:println(a); // @output 11.25
+
+    a = a2 + a3;
+    io:println(a); // @output 1
+
+    a = a2 + a4;
+    io:println(a); // @output -3.0
+
+    a = a2 + a5;
+    io:println(a); // @output 2.500000000000000000000000000000000E+1743
+
+    a = a2 + a6;
+    io:println(a); // @output -2.500000000000000000000000000000000E+1743
+
+    a = a3 + a4;
+    io:println(a); // @output 0
+
+    a = a3 + a5;
+    io:println(a); // @output 2.500000000000000000000000000000000E+1743
+
+    a = a3 + a6;
+    io:println(a); // @output -2.500000000000000000000000000000000E+1743
+}
+
+Test-Case: output
+Description: Test decimal addition with user-defined subtypes of decimal.
+Fail-Issue: ballerina-platform/ballerina-lang#32828
+Labels: additive-expr, decimal, DecimalFloatingPointNumber, DecimalNumber, FloatingPointTypeSuffix, module-const-decl,
+        module-type-defn, singleton-type, union-type
+
+const decimal A = 1.25;
+const decimal B = 2.0;
+
+type DecimalType1 A|B;
+
+function init() {
+    A a = 1.25;
+    decimal|DecimalType1 b = 25346.45d;
+
+    decimal x = a + b;
+    io:println(x); // @output 25347.70
+}
+
+Test-Case: output
+Description: Test whether contextually expected type of operands is decimal when the contextually expected
+             type of the additive-expr is decimal.
+Labels: additive-expr, decimal, DecimalNumber
+
+function init() {
+    decimal a = 10 + 20;
+    io:println(a); // @output 30
+}
+
+Test-Case: output
+Description: Test assigning the result of an additive expression to a compatible type.
+Labels: additive-expr, any, decimal, DecimalFloatingPointNumber, FloatingPointTypeSuffix, is-expr, string, union-type
+
+function init() {
+    decimal|string a = 10d + 20d;
+    io:println(a); // @output 30
+    io:println(a is decimal); // @output true
+
+    any b = 10d + 20d;
+    io:println(b); // @output 30
+    io:println(a is decimal); // @output true
+}
+
+Test-Case: output
+Description: Test assigning the result of an additive expression to var.
+Labels: additive-expr, decimal, DecimalFloatingPointNumber, FloatingPointTypeSuffix, is-expr, var
+
+function init() {
+    var a = 10d + 20d;
+    io:println(a); // @output 30
+    io:println(a is decimal); // @output true
+}
+
+Test-Case: output
+Description: Test using additive expression at module level.
+Labels: additive-expr, decimal, DecimalFloatingPointNumber, FloatingPointTypeSuffix, is-expr, module-const-decl,
+        module-var-decl, var
+
+var a = 10d + 20d;
+
+decimal b = 10d + 20d;
+
+const decimal c = 10d + 20d;
+
+function init() {
+    io:println(a); // @output 30
+    io:println(a is decimal); // @output true
+
+    io:println(b); // @output 30
+
+    io:println(c); // @output 30
+}
+
+Test-Case: output
+Description: Test decimal addition with positive floats of type decimal?.
```

Comment:
```suggestion
Description: Test decimal addition with positive decimals of type decimal?.
```

Reply:
Should this be updated anyway since the value may not be a positive decimal (a8 is nil).

Reply:
Fixed the descriptions wherever applicable.